common/src/main/java/com/github/khanshoaib3/minecraft_access/features/BiomeIndicator.java

```diff
@@ -6,45 +6,35 @@
 import net.minecraft.client.resource.language.I18n;
 import net.minecraft.registry.entry.RegistryEntry;
 import net.minecraft.world.biome.Biome;
+import org.jetbrains.annotations.Nullable;
 
 /**
  * Narrates the name of the biome when entering a different biome.
  */
 @Slf4j
 public class BiomeIndicator {
-    private String previousBiome;
+    @Nullable
+    private String previousBiome = null;
 
     public BiomeIndicator() {
-        previousBiome = "";
     }
 
     public void update() {
-        try {
-            MinecraftClient minecraftClient = MinecraftClient.getInstance();
-            if (minecraftClient == null) return;
-            if (minecraftClient.world == null) return;
-            if (minecraftClient.player == null) return;
-            if (minecraftClient.currentScreen != null) return;
-
-            RegistryEntry<Biome> var27 = minecraftClient.world.getBiome(minecraftClient.player.getBlockPos());
-            String name = I18n.translate(getBiomeName(var27));
-
-            /* Pre 1.19.3
-            Identifier id = minecraftClient.world.getRegistryManager().get(Registry.BIOME_KEY).getId(minecraftClient.world.getBiome(minecraftClient.player.getBlockPos()).value()); // Pre 1.19.3
-            if (id == null) {
-                log.error("\nUnable to detect biome!!\n");
-                return;
-            }
-            String name = I18n.translate("biome." + id.getNamespace() + "." + id.getPath());
-            */
-
-            if (!previousBiome.equalsIgnoreCase(name)) {
-                previousBiome = name;
-                MainClass.speakWithNarrator(I18n.translate("minecraft_access.other.biome", name), true);
-            }
-
-        } catch (Exception e) {
-            log.error("An error occurred while narrating biome.", e);
+        MinecraftClient minecraftClient = MinecraftClient.getInstance();
+        if (minecraftClient == null) return;
+        if (minecraftClient.world == null) return;
+        if (minecraftClient.player == null) return;
+        if (minecraftClient.currentScreen != null) return;
+
+        RegistryEntry<Biome> var27 = minecraftClient.world.getBiome(minecraftClient.player.getBlockPos());
+        String name = I18n.translate(getBiomeName(var27));
+
+        if (previousBiome == null) {
+            previousBiome = name;
+            MainClass.speakWithNarrator(I18n.translate("minecraft_access.other.biome", name), true);
+        } else if (!previousBiome.equalsIgnoreCase(name)) {
+            previousBiome = name;
+            MainClass.speakWithNarrator(I18n.translate("minecraft_access.other.biome", name), true);
         }
```

Comment:
Duplicated code. String.equalsIgnoreCase() will also return false if its argument is null so an explicit null check is superfluous.
```suggestion
        if (!name.equalsIgnoreCase(previousBiome)) {
            previousBiome = name;
            MainClass.speakWithNarrator(I18n.translate("minecraft_access.other.biome", name), true);
        }
```